dynamic_programming/fibonacci.py

```diff
@@ -1,50 +1,234 @@
+
+        # Extend sequence if needed
+        if n >= len(self.sequence):
+            difference = n - len(self.sequence) + 1
+            for _ in range(difference):
+                self.sequence.append(self.sequence[-1] + self.sequence[-2])
+
+        return self.sequence[n]
+
+    def reset(self) -> None:
+        """
+        Reset the sequence to its initial state.
+
+        Examples:
+            >>> fib = Fibonacci()
+            >>> fib.get(10)
+            [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]
+            >>> len(fib.sequence)
+            10
+            >>> fib.reset()
+            >>> fib.sequence
+            [0, 1]
+        """
+        self.sequence = [0, 1]
+
 
 def main() -> None:
+    """
+    Interactive CLI for generating Fibonacci numbers.
+
+    Allows users to input indices and get the corresponding Fibonacci sequence.
+    Supports commands: 'exit', 'quit', 'reset', 'help'
+    """
     print(
-        "Fibonacci Series Using Dynamic Programming\n",
-        "Enter the index of the Fibonacci number you want to calculate ",
-        "in the prompt below. (To exit enter exit or Ctrl-C)\n",
-        sep="",
+        "Fibonacci Series Using Dynamic Programming\n"
+        "Enter the index of the Fibonacci number you want to calculate\n"
+        "Commands: 'exit'/'quit' to exit, 'reset' to clear cache, 'help' for help\n"
     )
+
     fibonacci = Fibonacci()
 
     while True:
-        prompt: str = input(">> ")
-        if prompt in {"exit", "quit"}:
-            break
-
         try:
-            index: int = int(prompt)
-        except ValueError:
-            print("Enter a number or 'exit'")
-            continue
+            prompt: str = input(">> ").strip().lower()
+
+            if prompt in {"exit", "quit"}:
+                print("Goodbye!")
+                break
+            elif prompt == "reset":
+                fibonacci.reset()
+                print("Fibonacci sequence cache cleared.")
+                continue
+            elif prompt == "help":
+                print(
+                    "Commands:\n"
+                    "  <number>  - Get first N Fibonacci numbers\n"
+                    "  reset     - Clear the memoized sequence\n"
+                    "  help      - Show this help message\n"
+                    "  exit/quit - Exit the program\n"
+                )
+                continue
+            elif prompt == "":
+                continue
 
-        print(fibonacci.get(index))
+            try:
+                index: int = int(prompt)
+                if index < 0:
+                    print("Please enter a non-negative number.")
+                    continue
+
+                result = fibonacci.get(index)
+                if not result:
+                    print("[]")
+                else:
+                    print(f"First {index} Fibonacci numbers: {result}")
+                    if index > 0:
+                        print(f"The {index}th Fibonacci number is: {result[-1]}")
```

Comment:
Use the `get_nth` method here. `result[-1]` corresponds to the `nth - 1` element of the Fibonacci sequence. See the image below:

<img width="892" height="544" alt="image" src="https://github.com/user-attachments/assets/a53474dc-29a7-477c-b6ea-c4fc8e64baa5" />

```suggestion
                        print(f"The {index}th Fibonacci number is: {fibonacci.get_nth(index)}")
```